src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -334,102 +335,96 @@ const submitEncryptModeForm: ControllerHandler<
     }
   }
 
-  const submission = new EncryptSubmission({
+  const submissionContent = {
     form: form._id,
     authType: form.authType,
     myInfoFields: form.getUniqueMyInfoAttrs(),
     encryptedContent: incomingSubmission.encryptedContent,
     verifiedContent: verified,
     attachmentMetadata,
     version: req.body.version,
-  })
-
-  let savedSubmission
-  try {
-    savedSubmission = await submission.save()
-  } catch (err) {
-    logger.error({
-      message: 'Encrypt submission save error',
-      meta: {
-        action: 'onEncryptSubmissionFailure',
-        ...createReqMeta(req),
-      },
-      error: err,
-    })
-    return res.status(StatusCodes.BAD_REQUEST).json({
-      message:
-        'Could not send submission. For assistance, please contact the person who asked you to fill in this form.',
-      submissionId: submission._id,
-    })
   }
 
-  const submissionId = String(savedSubmission._id)
-  logger.info({
-    message: 'Saved submission to MongoDB',
-    meta: {
-      ...logMeta,
-      submissionId,
-      formId,
-    },
-  })
-
-  // Fire webhooks if available
-  // To avoid being coupled to latency of receiving system,
-  // do not await on webhook
-  const webhookUrl = form.webhook?.url
-  if (webhookUrl) {
-    void WebhookFactory.sendInitialWebhook(
-      submission,
-      webhookUrl,
-      !!form.webhook?.isRetryEnabled,
-    )
-  }
-
-  // Client secret for stripe payments if payments are enabled
-  let paymentClientSecret
+  // Handle submissions for payments forms
   if (
     form.payments_field?.enabled &&
     form.payments_channel?.channel === PaymentChannel.Stripe
   ) {
-    if (!form.payments_field.amount_cents) {
+    const amount = form.payments_field.amount_cents
+    // Step 1: Create payment without payment intent id and pending submission id.
+    if (
+      !amount ||
+      amount < paymentConfig.minPaymentAmountCents ||
+      amount > paymentConfig.maxPaymentAmountCents
+    ) {
       logger.error({
         message:
-          'Error when creating payment intent, amount is not a positive integer',
-        meta: {
-          submissionId,
-          ...logMeta,
-        },
+          'Error when creating payment: amount is not a positive integer',
```

Comment:
You forgot to change the error message accordingly 😅

Reply:
[91621ad](https://github.com/opengovsg/FormSG/pull/5999/commits/91621ad008f3cab0db82380699c8dea88d77e9b5)